usb_flash_drive_copy.py

```diff
@@ -0,0 +1,543 @@
+    pass
+
+
+@dataclass
+class Video:
+    filename: str
+    path: Path
+    copied: bool
+    delete: bool
+
+    @staticmethod
+    def from_dict(d: dict, src: Path) -> "Video":
+        filename = d["filename"]
+        copied = bool(re.search(r"yes|y|true|x|ja|j", d["copied"].lower()))
+        delete = bool(re.search(r"yes|y|true|x|ja|j", d["delete"].lower()))
+        return Video(filename, src / filename, copied, delete)
+
+    def __hash__(self) -> int:
+        return hash(self.path)
+
+    def to_dict(self) -> dict:
+        return {"filename": self.filename, "copied": self.copied, "delete": self.delete}
+
+    def __eq__(self, __o: object) -> bool:
+        """
+        A `Video` object is equal to another Video object if their path is the
+        same.
+        """
+        if not isinstance(__o, Video):
+            return NotImplemented
+        return self.path == __o.path
+
+
+class Led:
+    def __init__(self, led: PWMLED) -> None:
+        """Provides actions to interact with hardware LED.
+
+        Args:
+            led (PWMLED): Interface to hardware LED.
+        """
+
+        self._led = led
+
+    def blink(self, n: Union[int, None] = None, background: bool = True) -> None:
+        """Led blinking action.
+
+        Args:
+            n (Union[int, None], optional): Number of times to blink.
+            Defaults to `None` meaning forever.
+            background (bool, optional): Start as background thread.
+            If `False` return only when blink has finished when `n!=None`.
+            Defaults to `True`.
+        """
+        if not config.USE_LED:
+            return
+        self._led.blink(n=n, background=True)
+        time.sleep(2)
+
+    def turn_off(self) -> None:
+        """Turn off LED."""
+        if not config.USE_LED:
+            return
+        self._led.off()
+        time.sleep(2)
+
+    def turn_on(self) -> None:
+        """Turn on LED."""
+        if not config.USE_LED:
+            return
+        self._led.on()
+        time.sleep(2)
+
+
+class Button(Subject):
+    def __init__(self, name: str, button: GPIOButton) -> None:
+        """Automatically notifies list of observers of button state changes.
+
+        Args:
+            name (str): The name of the button.
+            button (GPIOButton): Interface to physical hardware button.
+        """
+        super().__init__()
+        self.name = name.upper()
+        self._button = button
+        self.is_active = button.is_active
+        self._check_button_is_active()
+        self._register_callbacks()
+
+    def _register_callbacks(self):
+        self._button.when_activated = self.on_pressed
+        self._button.when_deactivated = self.on_released
+        log.write(f"Register {self.name} button callbacks.", log.LogLevel.DEBUG)
+
+    def on_pressed(self):
+        """Notifies observers about button pressed event."""
+        log.write(f"{self.name} button pressed.", log.LogLevel.DEBUG)
+        self._check_button_is_active()
```

Comment:
That's a good question! At some point, I was treating the Button class in the context of a Power Button instead of a regular button. 
A power button release would indicate the system being shut down. Thus pressing the button again while still executing the shutdown procedure should ideally have no effect on the shutdown procedure. This logic/responsibility should be moved to the `OTCameraUsbCopier` class. I will change it accordingly

Reply:
I created a new issue in regards to recovering from partial copied files. 
The fix addressing preventing shutdown while copying files has been pushed to this PR's branch.